packages/components/src/search/autocompleters/variations.js

```diff
@@ -93,41 +93,69 @@ import ProductImage from '../../product-image';
  * attribute option strings.
  *
  * @param {Object} variation - variation returned by the api
- * @return {string} - variation name
+ * @param {Array} variation.attributes - attribute objects, with option property.
+ * @param {string} variation.name - name of variation.
+ * @return {string} - formatted variation name
  */
-function getVariationName( variation ) {
-	return variation.attributes.map( ( { option } ) => option ).join( ', ' );
+function getVariationName( { attributes, name } ) {
+	const separator =
+		window.wcSettings.variationTitleAttributesSeparator || ' - ';
```

Comment:
same here. Or is this the same function? Can it be placed somewhere reusable?

Reply:
`lib/async-requests` definitely duplicates logic from the autocompleters. I chose not to tackle that as this PR was already rather large 😅